providers/efa/efa.c

```diff
@@ -73,6 +73,10 @@ static struct verbs_context *efa_alloc_context(struct ibv_device *vdev,
 	ctx->max_llq_size = resp.max_llq_size;
 	pthread_spin_init(&ctx->qp_table_lock, PTHREAD_PROCESS_PRIVATE);
 
+	/* ah udata is mandatory for ah number retrieval */
+	if (!(ctx->cmds_supp_udata_mask & EFA_USER_CMDS_SUPP_UDATA_CREATE_AH))
```

Comment:
check_comp_mask()  for new code, please.

Reply:
check_comp_mask isn't quite appropriate here, it's not a standard case where we fail due to unknown bit set. This case is making sure that the bit is set.
It could be written like:

```
	if (check_comp_mask(ctx->cmds_supp_udata_mask,
			    ~EFA_USER_CMDS_SUPP_UDATA_CREATE_AH))
		goto err_free_spinlock;
```

But it makes the code more cryptic, especially since the usual check is for if (!check_comp_mask), not if (check_comp_mask).